File: 2023/src/day01/day01_test.go
package day01

import (
	at "aoc2023/common/aoc_testing"
	"os"
	"testing"
)

var example01, example02, input *string

func TestMain(m *testing.M) {
	example01 = at.MustRead("./data/example01.txt")
	example02 = at.MustRead("./data/example02.txt")
	input = at.MustRead("./data/input.txt")

	os.Exit(m.Run())
}

func TestD01p1(t *testing.T) {
	at.Run(t, part1, []at.Case[uint16]{
		{Name: "d01_p1_e01", Input: example01, Expected: 142},
		{Name: "d01_p1_i", Input: input, Expected: 54697},
	})
}

func TestD01p2(t *testing.T) {
	at.Run(t, part2, []at.Case[uint16]{
		{Name: "d01_p2_e02", Input: example02, Expected: 281},
		{Name: "d01_p2_i", Input: input, Expected: 54885},
	})
}

File: 2015/src/day01/day01_test.c
#include "day01.h"
#include "testing.h"
#include "utils.h"

#include <stdlib.h>

const test_entry* day01_tests[] = {
    TE("./data/example01.txt", "d01_p1_e01", one, 0),
    TE("./data/example02.txt", "d01_p1_e02", one, 0),
    TE("./data/example03.txt", "d01_p1_e03", one, 3),
    TE("./data/example04.txt", "d01_p1_e04", one, 3),
    TE("./data/example05.txt", "d01_p1_e05", one, 3),
    TE("./data/example06.txt", "d01_p1_e06", one, -1),
    TE("./data/example07.txt", "d01_p1_e07", one, -1),
    TE("./data/example08.txt", "d01_p1_e08", one, -3),
    TE("./data/example09.txt", "d01_p1_e09", one, -3),
    TE("./data/input.txt", "d01_p1_i", one, 74),
    TE("./data/example10.txt", "d01_p2_e10", two, 1),
    TE("./data/example11.txt", "d01_p2_e11", two, 5),
    TE("./data/input.txt", "d01_p2_i", two, 1795),
    nullptr
};

int day01_test_run(const test_entry* te) {
    char* input = file_to_string(te->file);
    int   r     = (te->p == one) ? day01_part1(input) : day01_part2(input);
    free(input);
    return r;
}
